### PR TITLE
Add tests to confirm that IOperation and SemanticModel agree for out variables

### PR DESCRIPTION
**Pull request by gafter**
Test-only change that confirms that #14295 is already fixed.

@AlekseyTs @dotnet/roslyn-compiler Please review.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs [hunk 1]
```diff
@@ -1015,6 +1015,7 @@ private static void VerifyModelForOutVarInNotExecutableCode(SemanticModel model,
 
             var typeInfo = model.GetTypeInfo(decl);
             Assert.Equal(expectedType, typeInfo.Type);
+            Assert.Equal(typeInfo.Type, model.GetOperationInternal(decl)?.Type);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: >typeInfo.Type [](start = 25, length = 13)

Please use expectedType instead.

**Comment by gafter** [comment]
please test windows_debug_vs-integration_prtest

**Comment by gafter** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by gafter** [comment]
retest windows_release_unit32_prtest please

**Pull request merged**